src/middlewares/piwik.js

```diff
@@ -0,0 +1,40 @@
+/* global Piwik */
+import {
+  PRE_UPLOAD_FILE
+} from '../actions'
```

Comment:
I agree, although I don't see a better way to do this. Keeping the actions somewhere else would mean the middleware needs an API to be configured, and would be practically empty. I don't mind the "extract and modify" approach in this case, but I'm open to other ways to structure this :)